### DISCUSSION TITLE
Hibernate Reactive Tracing + Stage.SessionFactory.openSession()

### DISCUSSION THREAD

**Discussion by s-af** [open-telemetry/opentelemetry-java-instrumentation#9474] [Q&A]
Hi,

I am trying to enable hibernate reactive tracing for project using Micronaut 4 + Hibernate Reactive + OTEL Java Agent. Unfortunately this combination is not working for me. I am able to receive Vert.x traces, but without link to parent span. Hibernate reactive traces are not generating at all. After some debugging I found that Micronaut always use Stage.SessionFactory.openSession() to obtain session, which is valid case according Hibernate documentation. From what I am seeing in source code OTEL Hibernate Reactive instrumentation instrument only methods Stage.SessionFactory.withSession() & withStatelessSession().

Are there any options how to make hibernate reactive traces work in combination with Stage.SessionFactory.openSession()? Is it some special case or just not implemented in OTEL?

Kindly correct me if my thoughts wrong.

**Comment by laurit** [comment]
It is not implemented. I have improved the instrumentation in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9486 You can try it out with https://github.com/open-telemetry/opentelemetry-java-instrumentation/actions/runs/6219392873#artifacts (download the build artifact and unzip it, it should contain the agent jar) and let us know whether it works any better.

**Comment by s-af** [comment]
Thank you, @laurit ! 
Now it looks better. I was able to get Vert.x traces linked to Netty http parent span. Unfortunately, still there is no Hibernate traces.

There is one more small issue with my setup. In case if there is no dependency to io.smallrye.reactive:mutiny, hibernate-reactive fails with error:
```
15:29:13.906 [default-nioEventLoopGroup-2-3] WARN  i.o.j.t.i.MuzzleMatcher [tid=9f19908d47d237c1d0db37201f834d5e sid=cfbb89db3ae0c7e0] - Instrumentation skipped, mismatched references were found: hibernate-reactive [class io.opentelemetry.javaagent.instrumentation.hibernate.reactive.v1_0.HibernateReactiveInstrumentationModule] on jdk.internal.loader.ClassLoaders$AppClassLoader@6f94fa3e
15:29:13.909 [default-nioEventLoopGroup-2-3] WARN  i.o.j.t.i.MuzzleMatcher [tid=9f19908d47d237c1d0db37201f834d5e sid=cfbb89db3ae0c7e0] - -- io.opentelemetry.javaagent.instrumentation.hibernate.reactive.v1_0.ContextOperator$ContextSubscriber:54 Missing class io.smallrye.mutiny.subscription.UniSubscription
15:29:13.910 [default-nioEventLoopGroup-2-3] WARN  i.o.j.t.i.MuzzleMatcher [tid=9f19908d47d237c1d0db37201f834d5e sid=cfbb89db3ae0c7e0] - -- io.opentelemetry.javaagent.instrumentation.hibernate.reactive.v1_0.ContextOperator:0 Missing class io.smallrye.mutiny.operators.UniOperator
15:29:13.910 [default-nioEventLoopGroup-2-3] WARN  i.o.j.t.i.MuzzleMatcher [tid=9f19908d47d237c1d0db37201f834d5e sid=cfbb89db3ae0c7e0] - -- io.opentelemetry.javaagent.instrumentation.hibernate.reactive.v1_0.ContextOperator:38 Missing class io.smallrye.mutiny.groups.UniSubscribe
15:29:13.912 [default-nioEventLoopGroup-2-3] WARN  i.o.j.t.i.MuzzleMatcher [tid=9f19908d47d237c1d0db37201f834d5e sid=cfbb89db3ae0c7e0] - -- io.opentelemetry.javaagent.instrumentation.hibernate.reactive.v1_0.ContextOperator:38 Missing class io.smallrye.mutiny.subscription.UniSubscriber
15:29:13.913 [default-nioEventLoopGroup-2-3] WARN  i.o.j.t.i.MuzzleMatcher [tid=9f19908d47d237c1d0db37201f834d5e sid=cfbb89db3ae0c7e0] - -- io.opentelemetry.javaagent.instrumentation.hibernate.reactive.v1_0.MutinySessionFactoryInstrumentation$ContextAdvice:39 Missing class io.smallrye.mutiny.Uni
```
Can be fixed by adding dependency to io.smallrye.reactive:mutiny

**Reply by laurit** [reply]
> Unfortunately, still there is no Hibernate traces.

Our hibernate-reactive instrumentation currently does not create any spans. Its purpose is to propagate context so that the span from vertx-sql would be linked with whatever called hibernate reactive. What kind of spans do you expect from the hibernate-reactive instrumentation?

> Can be fixed by adding dependency to io.smallrye.reactive:mutiny

I'll try to fix this

**Reply by s-af** [reply]
Ahh, I see. I compared hibernate-reactive instrumentation with regular hibernate instrumentation. Regular hibernate instrumentation send spans about opened sessions, DB commits, etc. So I was thinking hibernate-reactive instrumentation will do something similar. But if it not a case, then everything works well, thank you for your fix.
